### PR TITLE
Restructure core objects

### PR DESCRIPTION
**Pull request by dfaulken**
Complete with some tests:
screenshot from 2017-05-30 14-48-40

This would bump to version `0.1.0` due to the breaking change (working inside a row with `row` rather than `pdf`).

Major point of interest is `lib/prawn-rails-forms.rb`.

### CODE REVIEW

File: lib/prawn-rails-forms.rb [hunk 1]
```diff
@@ -3,13 +3,36 @@
...
   class FieldRow
-    attr_accessor :height, :units, :x, :y, :unit_width
-    
-    def initialize(height, units, x, y, unit_width)
-      @height, @units, @x, @y, @unit_width = 
-        height, units, x, y, unit_width
+    attr_accessor :document, :height, :units, :x, :y, :unit_width
+
+    def initialize(document,height, units, x, y, unit_width)
+      @document, @height, @units, @x, @y, @unit_width =
```
Comment (dfaulken) [inline lib/prawn-rails-forms.rb]: `@document` becomes an attribute of the `FieldRow` class, to which the `make_*_field` methods are sent.

**Review by werebus — APPROVED** [review]
Regarding @fermion's freedom-patching objection, I've commonly seen the idiom of defining your added methods in a `Module` and then opening the target class *just* to add an `include` (or `prepend`).  Functionally the same, except that your module shows up in the inheritance hierarchy for your monkey-patched class, which can help other programmers in debugging:

```ruby
module MyStringAdditions
  def shout
    upcase + "!!!"
  end
end

class String
  prepend MyStringAdditions

  def whisper
    "(" + downcase + ")"
  end
end

"Some string".method(:whisper) #=> #<Method: String#whisper>
"Some string".method(:shout)   #=> #<Method: String(MyStringAdditions)#shout>
```

**Review by Anbranin — APPROVED** [review]

**Review by fermion — APPROVED** [review]

**Comment by dfaulken** [comment]
Thanks everyone.

@werebus that's a great idea. OOC, there's probably no reason to consider that a not-backwards-compatible change, right?

**Comment by werebus** [comment]
I don't think so, the public interface stays the same.

**Pull request merged**